### PR TITLE
Add function to advertise accepted CA list for clients

### PR DESCRIPTION
**Pull request by rix0rrr**
Advertising the CA list properly will restrict the list of client certificates
that a browser will show to the user only to ones that will provide successful
authentication.

### CODE REVIEW

**Comment by rix0rrr** [comment]
Please help me understand how I broke the build?

File: openssl/src/ssl/mod.rs [hunk 1]
```diff
@@ -613,6 +613,19 @@ impl SslContext {
...
 
     #[allow(non_snake_case)]
+    /// Advertise the CAs from a file as being accepted for client certificates.
+    ///
+    /// This does not actually make the certificates trusted. You'll still need to call set_CA_file
+    /// for that.
+    pub fn advertise_client_CA_list<P: AsRef<Path>>(&mut self, file: P) -> Result<(), SslError> {
+        let file = CString::new(file.as_ref().as_os_str().to_str().expect("invalid utf8")).unwrap();
+        wrap_ssl_result(unsafe {
+            let names = ffi::SSL_load_client_CA_file(file.as_ptr() as *const _);
```
Comment (sfackler) [inline openssl/src/ssl/mod.rs]: This'll need to be run through 2 rounds of error checking, or we'll pass a null pointer to ssl_ctx_set_client_ca_list

**Comment by sfackler** [comment]
Think you'll have a chance to make the small tweak?

**Comment by rix0rrr** [comment]
Yeah, sorry. Will do sometime soon.

File: openssl/src/ssl/mod.rs [hunk 2]
```diff
@@ -613,6 +613,24 @@ impl SslContext {
     }
 
     #[allow(non_snake_case)]
+    /// Advertise the CAs from a file as being accepted for client certificates.
+    ///
+    /// This does not actually make the certificates trusted. You'll still need to call set_CA_file
+    /// for that.
+    pub fn advertise_client_CA_list<P: AsRef<Path>>(&mut self, file: P) -> Result<(), SslError> {
```
Comment (sfackler) [inline openssl/src/ssl/mod.rs]: Can this name get changed to `set_client_CA_list` to match the OpenSSL function name?

**Comment by sfackler** [comment]
Needs a rebase

**Comment by sfackler** [comment]
Implemented in 1edb6f682eaa728871e39aa41735a668fcc7447c

**Pull request closed without merging**